libraries/classes/Import/SimulateDml.php

```diff
@@ -132,12 +132,11 @@ private function getSimulatedUpdateQuery(Parser $parser, UpdateStatement $statem
         $diff = [];
         foreach ($statement->set as $set) {
             $columns[] = $set->column;
-            $notEqualOperator = ' <> ';
             if (strtoupper($set->value) === 'NULL') {
-                $notEqualOperator = ' IS NOT ';
+                $diff[] = $set->column . ' IS NOT ' . $set->value;
+            } else {
+                $diff[] = $set->column . ' <> ' . $set->value . ' OR ' . $set->column . ' IS NULL';
             }
```

Comment:
Some love  for the [MySQL `<=>` operator](https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#operator_equal-to)?
```suggestion
            $diff[] = 'NOT ' . $set->column . ' <=> ' . $set->value;
```

Reply:
It broke my brain 😂
Thanks for sharing it

Reply:
Thank you @MoonE ! I kept the additional parentheses that you have added it seems to keep the `value` safe. but are there any special conditions without them that ruin the query?

Reply:
This can happen with an update query like 
```sql
UPDATE a SET b = c OR d WHERE e OR f
```
It would generate `NOT b <=> c OR d` instead of  `NOT b <=> (c OR d)`

With that in mind the original condition should also be  set in parenthesis:
```patch
        if (! empty($diff)) {
-            $where .= ' AND (' . implode(' OR ', $diff) . ')';
+            $where = '(' . $where . ') AND (' . implode(' OR ', $diff) . ')';
        }
``` 
to generate the condition
`WHERE (e OR f) AND (NOT b <=> (c OR d))` instead of 
`WHERE e OR f AND (NOT b <=> (c OR d))`